File: sg_cdez_api/src/main/java/com/cdez/sg_cdez_api/service/MedicamentoService.java
package com.cdez.sg_cdez_api.service;

import com.cdez.sg_cdez_api.dto.request.MedicamentoCreateRequest;
import com.cdez.sg_cdez_api.dto.request.MedicamentoUpdateRequest;
import com.cdez.sg_cdez_api.dto.response.MedicamentoResponse;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
public interface MedicamentoService {
    List<MedicamentoResponse> listarMedicamentosPorAdulto(UUID adultoId);
    MedicamentoResponse crearMedicamentos(MedicamentoCreateRequest request, UUID adultoId);
    MedicamentoResponse actualizarMedicamentos(MedicamentoUpdateRequest request, UUID adultoId);
    void desactivarMedicamentos(UUID medicamentos);
    void activarMedicamentos(UUID medicamentos);
}

File: sg_cdez_web/src/services/medicamentoService.ts
import { apiClient } from "../utils/apiHelper";

import type {
  MedicamentoCreateRequest,
  MedicamentoResponse,
  MedicamentoUpdateRequest,
} from "./interfaces/medicamentoInterface";

export async function listarMedicamentosPorAdulto(
  adultoId: string,
): Promise<MedicamentoResponse[]> {
  const response = await apiClient.post<MedicamentoResponse[]>(
    `/medicamentos/listarMedicamentos/${adultoId}`,
  );

  return response.data;
}

export async function registrarMedicamento(
  adultoId: string,
  request: MedicamentoCreateRequest,
): Promise<MedicamentoResponse> {
  const response = await apiClient.post<MedicamentoResponse>(
    `/medicamentos/crearMedicamentos/${adultoId}`,
    request,
  );

  return response.data;
}

export async function actualizarMedicamento(
  adultoId: string,
  request: MedicamentoUpdateRequest,
): Promise<MedicamentoResponse> {
  const response = await apiClient.put<MedicamentoResponse>(
    `/medicamentos/actualizarMedicamentos/${adultoId}`,
    request,
  );

  return response.data;
}

export async function desactivarMedicamento(
  medicamentoId: string,
): Promise<void> {
  await apiClient.post(`/medicamentos/desactivarMedicamentos/${medicamentoId}`);
}
